.github/workflows/test.yml

```
@@ -5,9 +5,34 @@ on:
   workflow_call:
 
 jobs:
+  pre-rails-tests:
+    name: Pre Rails Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Determine if pre-rails tests are needed
+        id: check-rails-relevant-changes
+        run: |
+          if [ ${{ github.event_name }} == "pull_request" ]; then
+            git fetch origin ${{ github.event.pull_request.head.ref }}
+            RELEVANT_DIRS=$(git diff ${{ github.event.pull_request.base.sha }} origin/${{ github.event.pull_request.head.ref }}  --name-only)
+            echo "Relevant changes for rails: $RELEVANT_DIRS"
+            if [ -z "$(echo "$RELEVANT_DIRS" | egrep -v "^(.github|adr|aws|docs|public|storage|terraform|tests|tests-examples)/")" ]; then
```

```suggestion
            if [ -z "$(echo "$RELEVANT_DIRS" | egrep -v "^(.github|adr|aws|docs|storage|terraform|tests|tests-examples)/")" ]; then
```

`public` contains files that are served by Rails so we would want to run the tests if any of these change.

---

Will update